### PR TITLE
Add a randomized test for the variable-looping-vus requirements

### PR DESCRIPTION
**Pull request by na--**
I gave up on completely understanding all of the corner cases of https://github.com/loadimpact/k6/pull/1392, or on simplifying it... for now... :sweat_smile: But here's an extra test that hopefully validates that however we segment a `variable-looping-vus` executor, it will always execute what it would have executed if it was running on a single machine. 

Whether that execution is completely correct is another topic :smile: My manual tests didn't find any issues there, but I would still want some easier to follow implementation eventually, maybe when we adapt it to data partitioning.

### CODE REVIEW

File: lib/executor/variable_looping_vus_test.go [hunk 1]
```diff
@@ -1002,3 +1004,115 @@ func TestSegmentedIndex(t *testing.T) {
...
+		c.StartVUs = null.IntFrom(r.Int63n(maxVUs))
+		c.Stages = stages
+		return c
+	}
+
+	subtractChildSteps := func(t *testing.T, parent, child []lib.ExecutionStep) {
+		t.Logf("subtractChildSteps()")
+		for i := 0; i < len(child); i++ {
+			t.Logf("	child planned VUs for time offset %s: %d", child[i].TimeOffset, child[i].PlannedVUs)
+		}
```
Comment (MStoykov) [inline lib/executor/variable_looping_vus_test.go:1067]: this would've been more readable as 
```suggestion
		for _, step := range child {
			t.Logf("	child planned VUs for time offset %s: %d", step.TimeOffset, step.PlannedVUs)
		}
```

File: lib/executor/variable_looping_vus_test.go [hunk 2]
```diff
@@ -1002,3 +1004,115 @@ func TestSegmentedIndex(t *testing.T) {
...
+			if parent[p].TimeOffset == child[c].TimeOffset {
+				t.Logf("Setting sub to %d at t=%s", child[c].PlannedVUs, child[c].TimeOffset)
+				sub = child[c].PlannedVUs
+				if c != len(child)-1 {
+					c++
+				}
+			}
+			t.Logf("Subtracting %d VUs (out of %d) at t=%s", sub, parent[p].PlannedVUs, parent[p].TimeOffset)
+			parent[p].PlannedVUs -= sub
+		}
```
Comment (MStoykov) [inline lib/executor/variable_looping_vus_test.go:1082]: Can we also do it with a `range` and add a comment that we need to go through all the parent steps to remove the child ones so we can't break ... as that took me a while to figure out

**Comment by na--** [comment]
> I was thinking of doing it by adding up all the segments steps 
This would definitely be more complicated. I was going to refactor `GetFullExecutionRequirements()`, to use it here as well, but it ended up nasty, and wasn't worth it just for this test, so I sort of implemented the reverse of it, which was much simpler.

File: lib/executor/variable_looping_vus_test.go (hunk 2 above)
Comment (na--) [reply lib/executor/variable_looping_vus_test.go]: Resolved by https://github.com/loadimpact/k6/pull/1413/commits/d47a00bec8d8ccbe4f616982c78b496ce78da136

File: lib/executor/variable_looping_vus_test.go (hunk 1 above)
Comment (na--) [reply lib/executor/variable_looping_vus_test.go]: Resolved by https://github.com/loadimpact/k6/pull/1413/commits/d47a00bec8d8ccbe4f616982c78b496ce78da136

**Comment by na--** [comment]
@MStoykov I think we should first merge this in your branch, despite the unrelated failing CDNJS test, and then you can rebase on `new-schedulers`, where the test has been silenced?

File: lib/executor/variable_looping_vus_test.go [hunk 3]
```diff
@@ -1002,3 +1004,117 @@ func TestSegmentedIndex(t *testing.T) {
...
+				t.Errorf("ERR Could not match child offset %s with any parent time offset", child[ci].TimeOffset)
+			}
+			if p.TimeOffset == child[ci].TimeOffset {
+				t.Logf("Setting sub to %d at t=%s", child[ci].PlannedVUs, child[ci].TimeOffset)
+				sub = child[ci].PlannedVUs
+				if ci != len(child)-1 {
+					ci++
+				}
+			}
+			t.Logf("Subtracting %d VUs (out of %d) at t=%s", sub, p.PlannedVUs, p.TimeOffset)
```
Comment (imiric) [inline lib/executor/variable_looping_vus_test.go:1085]: This test is very noisy when run with `-v`, which I usually enable to see the full output, but now this test prints ~80K lines, forcing me to use `tee`, etc. :-/

Can we use a debug logger here?

Comment (MStoykov) [reply lib/executor/variable_looping_vus_test.go:1085]: I am for commenting all the `Logf` lines and leaving only the random seed one as that is needed to get it reproduced - if you get an error on this, trust me you would want to be able to reproduce it ... the size of the output is way too big to just read it in circleci for example
